config/clients/python/template/oauth2.mustache

```diff
@@ -32,25 +56,59 @@ class OAuth2Client:
         Perform OAuth2 and obtain token
         """
         configuration = self._credentials.configuration
+
         token_url = f'https://{configuration.api_issuer}/oauth/token'
+
         post_params = {
             'client_id': configuration.client_id,
             'client_secret': configuration.client_secret,
             'audience': configuration.api_audience,
             'grant_type': "client_credentials",
         }
+
         headers = urllib3.response.HTTPHeaderDict({'Accept': 'application/json', 'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'openfga-sdk (python) {{packageVersion}}'})
-        raw_response = await client.POST(token_url, headers=headers, post_params=post_params);
-        if 200 <= raw_response.status <= 299:
-            try:
-                api_response = json.loads(raw_response.data)
-            except:
-                raise AuthenticationError(http_resp=raw_response)
-            if not api_response.get('expires_in') or not api_response.get('access_token'):
-                raise AuthenticationError(http_resp=raw_response)
-            self._access_expiry_time = datetime.now() + timedelta(seconds=int(api_response.get('expires_in')))
-            self._access_token = api_response.get('access_token')
-        else:
+
+        max_retry = (
+            self.configuration.retry_params.max_retry
+            if (
+                self.configuration.retry_params is not None
+                and self.configuration.retry_params.max_retry is not None
+            )
+            else 0
+        )
+
+        min_wait_in_ms = (
+            self.configuration.retry_params.min_wait_in_ms
+            if (
+                self.configuration.retry_params is not None
+                and self.configuration.retry_params.min_wait_in_ms is not None
+            )
+            else 0
+        )
+
+        for attempt in range(max_retry + 1):
+            raw_response = await client.POST(
+                token_url, headers=headers, post_params=post_params
+            )
+
+            if 500 <= raw_response.status <= 599 or raw_response.status == 429:
```

Comment:
It seems like we're raising Auth errors on 404?

Can you check:
https://github.com/openfga/js-sdk/blob/main/common.ts#L155-L175
It doesn't handle 501s, but it covers the rest of the cases.

For the special cases of oauth token exchange, we then catch the error, whatever it is and mark it as "FgaApiAuthenticationError"
https://github.com/openfga/js-sdk/blob/main/credentials/credentials.ts#L163-L169